src/heatshrink.pyx

```diff
@@ -0,0 +1,237 @@
+import array
+import numbers
+cimport cython
+from cpython cimport array
+from libc.stdint cimport uint8_t
+
+cimport cheatshrink
+
+MIN_WINDOW_SZ2 = cheatshrink.HEATSHRINK_MIN_WINDOW_BITS
+MAX_WINDOW_SZ2 = cheatshrink.HEATSHRINK_MAX_WINDOW_BITS
+DEFAULT_WINDOW_SZ2 = 11
+
+MIN_LOOKAHEAD_SZ2 = cheatshrink.HEATSHRINK_MIN_LOOKAHEAD_BITS
+DEFAULT_LOOKAHEAD_SZ2 = 4
+
+DEFAULT_INPUT_BUFFER_SIZE = 2048
+
+
+cdef validate_bounds(val, name, min=None, max=None):
+    """
+    Ensure that `val` is larger than `min` and smaller than
+    `max`. Throws `ValueError` if those constraints are not met.
+
+    Will throw `ValueError` if both `min` and `max` are None.
+    """
+    if min is None and max is None:
+        raise ValueError("Expecting either a min or max parameter")
+
+    if not isinstance(val, numbers.Number):
+        msg = 'Expected number, got {}'
+        raise TypeError(msg.format(msg.__class__.__name__))
+
+    if min and val < min:
+        msg = "{} must be > {}".format(name, min)
+    elif max and val > max:
+        msg = "{} must be < {}".format(name, max)
+    else:
+        msg = ''
```

Comment:
I'd rather initialize msg as None and remove the else statement, but this might be just the matter of taste